bsp/src/org/jetbrains/bsp/project/resolver/BspProjectResolver.scala

```diff
@@ -113,11 +114,14 @@ class BspProjectResolver extends ExternalSystemProjectResolver[BspExecutionSetti
 
   private def runImport(workspace: File, executionSettings: BspExecutionSettings)
                        (implicit reporter: BuildReporter) = {
-    val notifications: NotificationCallback = {
+    def notifications(implicit reporter: BuildReporter): NotificationAggregator[BuildMessages] =
+    (messages, notification) => notification match {
       case BspNotifications.LogMessage(params) =>
-        // TODO use params.id for tree structure?
         reporter.log(params.getMessage)
+        messages.addWarning(params.getMessage)
```

Comment:
@jastice I really don't like this code (adding a warning and then reading _.warnings) but I couldn't find any better way to utilize BuildMessages in this case. The messages are streamed line-by-line here from bloop.

Reply:
how about adding a `log` field to `BuildMessages` instead?

Reply:
Actually, I'll just do this change